src/mlir/generator.cc

```diff
@@ -31,29 +31,31 @@ namespace mlir::verona
   using namespace ASTInterface;
 
   // ===================================================== Public Interface
-  void Generator::readAST(const ::ast::Ast& ast)
+  llvm::Error Generator::readAST(const ::ast::Ast& ast)
   {
     // Parse the AST with the rules below
-    parseModule(ast);
+    if (auto err = parseModule(ast))
+      return err;
 
     // On error, dump module for debug purposes
     if (mlir::failed(mlir::verify(*module)))
     {
       module->dump();
-      throw std::runtime_error("MLIR verification failed from Verona file");
+      return runtimeError("MLIR verification failed from Verona file");
```

Comment:
First caller and now I see how this API is meant to be used.  

If this string is presented to the user, we should provide an enum value rather than a string and map that to a string later, so that it can be localised without needing to modify huge numbers of unrelated source files.  The parser in the interpreter already does this.

Reply:
We should do that as part of a localisation effort. I'd rather think about it and reach a consensus than rush a half-baked implementation. Internationalisation tends to spiral out of control if done badly initially.